src/flownet/ahm/_run_ahm.py

```diff
@@ -22,6 +23,69 @@
 from ..data import FlowData
 
 
+def _from_regions_to_flow_tubes(
+    network: NetworkModel,
+    field_data: FlowData,
+    ti2ci: pd.DataFrame,
+    name: str,
+) -> List[int]:
+    """
+        The function loops through each cell in a flow tube, and checks what 'name' region the
+        corresponding position (cell midpoint) in the data source simulation model has. If different
+        cells in one flow tube are located in different 'name' regions of the original model, the mode is used.
+        If flow tubes are entirely outside of the data source simulation grid,
+        the 'name' region closest to the midpoint of the flow tube is used.
+
+    Args:
+        network: FlowNet network instance
+        field_data: FlowData class with information from simulation model data source
+        ti2ci: A dataframe with index equal to tube model index, and one column which equals cell indices.
+        name: The same of the region parameter
+
+    Returns:
+        A list with values for 'name' region for each tube in the FlowNet model
+    """
+    df_regions = []
+
+    x_mid = network.grid[["x0", "x1", "x2", "x3", "x4", "x5", "x6", "x7"]].mean(axis=1)
+    y_mid = network.grid[["y0", "y1", "y2", "y3", "y4", "y5", "y6", "y7"]].mean(axis=1)
+    z_mid = network.grid[["z0", "z1", "z2", "z3", "z4", "z5", "z6", "z7"]].mean(axis=1)
```

Comment:
Maybe make a property out of this.

You can copy that from my PR:

https://github.com/equinor/flownet/pull/155/files